timelock-server-benchmark-client/src/main/java/com/palantir/atlasdb/timelock/benchmarks/benchmarks/AbstractRangeScanBenchmark.java

```diff
@@ -0,0 +1,188 @@
+
+        while (numWritten < numRows) {
+            int numToWrite = Math.min(numRows - numWritten, batchSize);
+
+            txnManager.runTaskWithRetry(txn -> {
+                Map<Long, byte[]> values = Maps.newHashMapWithExpectedSize(numToWrite);
+                for (int i = 0; i < numToWrite; i++) {
+                    values.put(key.getAndIncrement(), data);
+                }
+
+                writeValues(txn, values);
+                return null;
+            });
+
+            numWritten += numToWrite;
+        }
+    }
+
+    @Override
+    protected final void performOneCall() {
+        List<byte[]> results = txnManager.runTaskReadOnly(txn -> getRange(txn, 0L, numRows));
+
+        Preconditions.checkState(results.size() == numRows);
+        for (byte[] resultData : results) {
+            Preconditions.checkState(resultData.length == dataSize);
+        }
+    }
+
+    protected abstract List<byte[]> getRange(Transaction txn, long startInclusive, long endExclusive);
+
+    protected abstract void writeValues(Transaction txn, Map<Long, byte[]> valuesByKey);
+
+    @Override
+    public Map<String, Object> getExtraParameters() {
+        return ImmutableMap.of("numRows", numRows, "dataSize", dataSize);
+    }
+
+    public static void main(String[] args) {
+        Metadata metadata = new Metadata("foo");
+        byte[] data = serialize(metadata);
+        deserialize(data);
+    }
+
+    private void recordMetadata() {
+        txnManager.runTaskWithRetry(txn -> {
+            MetadataTable table = BenchmarksTableFactory.of().getMetadataTable(txn);
+
+            Metadata metadata = new Metadata(bucket);
+            table.putData(MetadataTable.MetadataRow.of(getKeyForParameters()), serialize(metadata));
+            return null;
+        });
+    }
+
+    private Optional<String> getExistingBucketForParameters() {
+        return txnManager.runTaskReadOnly(txn -> {
+            MetadataTable table = BenchmarksTableFactory.of().getMetadataTable(txn);
+
+            return table.getRow(MetadataTable.MetadataRow.of(getKeyForParameters()))
+                    .map(MetadataTable.MetadataRowResult::getData)
+                    .map(AbstractRangeScanBenchmark::deserialize)
+                    .map(mdata -> mdata.bucket);
+        });
+    }
+
+    private static byte[] serialize(Metadata metadata) {
+        try {
+            return MAPPER.writeValueAsBytes(metadata);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Metadata deserialize(byte[] blob) {
+        try {
+            return MAPPER.readValue(blob, Metadata.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String getKeyForParameters() {
+        return getClass().getSimpleName() + "."
+                + dataSize + "."
+                + numRows;
+
+    }
+
+    private static final class Metadata {
```

Comment:
nit: immutables would give you equals/hashCode for free, in case anyone wanted to compare to `Metadata`s.